src/components/pages/devices/flyouts/deviceJobs/deviceJobTags.js

```diff
@@ -1,20 +1,311 @@
+          return (update(newState, {
+            commonTags: { $push: [{ name: tagName, value, type }] }
+          }));
+        },
+        initialState
+      )
+      // Update the component state
+      .subscribe(newState => {
+        this.setState(newState);
+      });
+  }
+
+  formIsValid() {
+    return [
+      this.jobNameLink
+    ].every(link => !link.error);
+  }
+
+  apply = (event) => {
+    event.preventDefault();
+    if (this.formIsValid()) {
+      this.setState({ isPending: true });
+
+      const { devices } = this.props;
+      const { commonTags, deletedTags } = this.state;
+      const request = toSubmitTagsJobRequestModel(devices, this.state);
+
+      this.subscription = IoTHubManagerService.sumbitJob(request)
+        .subscribe(
+          ({ jobId }) => {
+            this.setState({ jobId, successCount: devices.length, isPending: false, changesApplied: true, foo: Date.now() });
+            this.props.updateTags({ deviceId: devices.map(({ id }) => id), commonTags, deletedTags });
+          },
+          errorResponse => {
+            this.setState({ error: errorResponse.errorMessage, isPending: false, changesApplied: true });
+          }
+        );
+    }
+  }
+
+  getSummaryMessage() {
+    const { t } = this.props;
+    const { isPending, changesApplied } = this.state;
+
+    if (isPending) {
+      return t('devices.flyouts.jobs.pending');
+    } else if (changesApplied) {
+      return t('devices.flyouts.jobs.applySuccess');
+    } else {
+      return t('devices.flyouts.jobs.affected');
+    }
+  }
+
+  addTag = () => this.tagsLink.set([...this.tagsLink.value, newTag()]);
+
+  deleteTag = (index) =>
+    (evt) => {
+      this.setState(update(this.state, {
+        commonTags: { $set: this.tagsLink.value.filter((_, idx) => index !== idx) },
+        deletedTags: { $push: [this.tagsLink.value[index].name] }
+      }));
+    };
+
+  render() {
+    const {
+      t,
+      onClose,
+      devices
+    } = this.props;
+    const {
+      isPending,
+      error,
+      successCount,
+      changesApplied,
+      commonTags = []
+    } = this.state;
+
+    const summaryCount = changesApplied ? successCount : devices.length;
+    const completedSuccessfully = changesApplied && successCount === devices.length;
+    const summaryMessage = this.getSummaryMessage();
+    const typeOptions = [
+      {
+        value: TagJobConstants.numberType,
+        label: t('devices.flyouts.jobs.tags.typeNumber')
+      },
+      {
+        value: TagJobConstants.stringType,
+        label: t('devices.flyouts.jobs.tags.typeString')
+      }
+    ];
+
+    // Link these values in render because they need to update based on component state
+    const tagLinks = this.tagsLink.getLinkedChildren(tagLink => {
+      const name = tagLink.forkTo('name')
+        .check(Validator.notEmpty, () => this.props.t('devices.flyouts.jobs.validation.required'));
```

Comment:
May use `.check(Validator.notEmpty, this.props.t('devices.flyouts.jobs.validation.required'));` instead of `.check(Validator.notEmpty, () => this.props.t('devices.flyouts.jobs.validation.required'));`

Reply:
Hugh is correct. Since this is in the render function it will be remade when the props change. You only need to wrap the message in a function when you need to grab the props in a function closure.